eigrpd/eigrpd.c

```diff
@@ -160,14 +159,15 @@ static struct eigrp *eigrp_new(const char *AS, vrf_id_t vrf_id)
 	eigrp->passive_interface_default = EIGRP_IF_ACTIVE;
 	eigrp->networks = eigrp_topology_new();
 
-	if ((eigrp_socket = eigrp_sock_init()) < 0) {
+	eigrp->fd = eigrp_sock_init(vrf_lookup_by_id(vrf_id));
+
+	if (eigrp->fd < 0) {
 		flog_err_sys(
 			EC_LIB_SOCKET,
 			"eigrp_new: fatal error: eigrp_sock_init was unable to open a socket");
 		exit(1);
```

Comment:
2 questions:
- as the vrf hooks are empty, there is some work remaining that should prevent from exiting, if vrf is not ready yet.
- is there a plan to use one single socket for all vrfs (in case vrf-lite) , or not ( the former case, that means vrf socket call should be done once in case vrf_backend().is not netns)